### PR TITLE
Issue 8882: Fixes permissions of pinned posts

### PR DESCRIPTION
**Pull request by annando**
Fixes https://github.com/friendica/friendica/issues/8882

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -201,18 +201,10 @@ public static function selectPinned(int $uid, array $selected = [], array $condi
...
-			$first_key = key($condition);
-			if (!is_int($first_key)) {
-				$condition['iid'] = $pinned;
-			} else {
-				$values_string = substr(str_repeat("?, ", count($pinned)), 0, -2);
-				$condition[0] = '(' . $condition[0] . ") AND `iid` IN (" . $values_string . ")";
-				$condition = array_merge($condition, $pinned);
-			}
+			$condition = ['iid' => $pinned];
 		}
```
Comment (MrPetovan) [inline src/Model/Item.php:208]: You can merge a condition with an empty array and replace this whole block with the following:
```php
$condition = DBA::mergeConditions($condition, ['iid' => $pinned]);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php:208]: It is some (slight) performance advantage not calling that function, since it doesn't have any checks for this condition.

Comment (MrPetovan) [reply src/Model/Item.php:208]: I don't think we are at that level of optimization, I'd rather reduce the number of conditions and code lines in general.

File: src/Database/DBA.php [hunk 2]
```diff
@@ -669,6 +669,10 @@ public static function collapseCondition(array $condition)
 	 */
 	public static function mergeConditions(array ...$conditions)
 	{
+		if (count($conditions) == 1) {
+			return current($conditions);
+		}
```
Comment (MrPetovan) [inline src/Database/DBA.php:674]: This isn't the condition you want in this case. Instead, you should add the following condition in the following `foreach`:
```php
if (!$condition) {
    continue;
}
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**